superlu build fails: csh install.csh: csh: Command not found

I'm seeing `superlu@4.3` builds fail if `csh` is not available.

I will submit a PR making `tcsh` a build dependency of `superlu` unless someone objects.

System info:
* `spack@develop` (cf37e9276d826a4607afcc307f765c0708e83311 from `Fri Dec 11 16:54:11 2020 -0700`)
* `RHEL 7` for `x86_64`
* `GCC 8.1.0`
* Docker image: `ecpe4s/rhel7-runner-x86_64:2020-12-01`

Concrete spec: [superlu.spec.yaml.txt](https://github.com/spack/spack/files/5691392/superlu.spec.yaml.txt)
Build log: [superlu-build-out.txt](https://github.com/spack/spack/files/5691393/superlu-build-out.txt)

```
$> spack install -f ./superlu.spec.yaml
...
==> Installing superlu-4.3-qlkyqd2ey7zyququhslfypwda7joupai
==> No binary for superlu-4.3-qlkyqd2ey7zyququhslfypwda7joupai found: installing from source
==> Using cached archive: /opt/spack/var/spack/cache/_source-cache/archive/16/169920322eb9b9c6a334674231479d04df72440257c17870aaa0139d74416781.tar.gz
==> superlu: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make'

2 errors found in build log:
     19    /opt/spack/lib/spack/env/gcc/gcc -c -fPIC superlu_timer.c
     20    /opt/spack/lib/spack/env/gcc/gcc -O3 -fPIC -c timertst.c
     21    /opt/spack/lib/spack/env/gcc/gcc -fPIC -o testtimer superlu_timer.o timertst.o
     22    Testing machines parameters and timer
     23    csh install.csh
     24    make[1]: csh: Command not found
  >> 25    make[1]: *** [install.out] Error 127
     26    make[1]: Leaving directory `/tmp/root/spack-stage/spack-stage-superlu-4.3-qlkyqd2ey7zyququhslfypwda7joupai/spack-src/INSTALL'
  >> 27    make: *** [install] Error 2

See build log for details:
  /tmp/root/spack-stage/spack-stage-superlu-4.3-qlkyqd2ey7zyququhslfypwda7joupai/spack-build-out.txt
```

@adamjstewart @jrood-nrel

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -18,6 +18,7 @@ class Superlu(Package):
     variant('pic',    default=True,
             description='Build with position independent code')
 
+    depends_on('tcsh', type='build')
     depends_on('cmake', when='@5.2.1:', type='build')
     depends_on('blas')
 

